ksql-functional-tests/src/test/resources/query-validation-tests/array-functions.json

```diff
@@ -0,0 +1,110 @@
+        "CREATE STREAM test (v DOUBLE) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE STREAM OUTPUT AS SELECT ROUND(v) as R0, ROUND(v, 0) as R00, ROUND(v, 1) as R1, ROUND(v, 2) as R2, ROUND(v, 10) as R10, ROUND(v , -1) as 1R , ROUND(v , -2) as 2R  FROM test;"
+      ],
+      "inputs": [
+        {"topic": "test_topic", "value": {"v" : 1.2}},
+        {"topic": "test_topic", "value": {"v" : 1.7}},
+        {"topic": "test_topic", "value": {"v" : 1.5}},
+        {"topic": "test_topic", "value": {"v" : 3}},
+        {"topic": "test_topic", "value": {"v" : 1.234567}},
+        {"topic": "test_topic", "value": {"v" : 0}},
+        {"topic": "test_topic", "value": {"v" : 111}},
+        {"topic": "test_topic", "value": {"v" : null}},
+        {"topic": "test_topic", "value": {"v" : -1.1}},
+        {"topic": "test_topic", "value": {"v" : -1.5}},
+        {"topic": "test_topic", "value": {"v" : -1.7}},
+        {"topic": "test_topic", "value": {"v" : 12345.678}},
+        {"topic": "test_topic", "value": {"v" : -12345.678}}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "value": {"R0" : 1, "R00" : 1.0, "R1" :  1.2, "R2" : 1.2,  "R10" :  1.2, "1R" :  0.0 ,"2R" :  0.0 }},
+        {"topic": "OUTPUT", "value": {"R0" : 2, "R00" : 2.0, "R1" :  1.7, "R2" : 1.7,  "R10" :  1.7, "1R" :  0.0 ,"2R" :  0.0 }},
+        {"topic": "OUTPUT", "value": {"R0" : 2, "R00" : 2.0, "R1" :  1.5, "R2" : 1.5,  "R10" :  1.5, "1R" :  0.0 ,"2R" :  0.0 }},
+        {"topic": "OUTPUT", "value": {"R0" : 3, "R00" : 3.0, "R1" :  3.0, "R2" : 3.0,  "R10" :  3.0, "1R" :  0.0 ,"2R" :  0.0 }},
+        {"topic": "OUTPUT", "value": {"R0" : 1, "R00" : 1.0, "R1" :  1.2, "R2" : 1.23, "R10" :  1.234567,"1R" :  0.0 ,"2R" :  0.0 }},
+        {"topic": "OUTPUT", "value": {"R0" : 0, "R00" : 0.0, "R1" :  0.0, "R2" : 0.0,  "R10" :  0.0, "1R" :  0.0 ,"2R" :  0.0 }},
+        {"topic": "OUTPUT", "value": {"R0" : 111, "R00" : 111.0, "R1" :  111.0, "R2" : 111.0,  "R10" :  111.0, "1R" :  110.0 ,"2R" :  100.0 }},
+        {"topic": "OUTPUT", "value": {"R0" : null, "R00" : null, "R1" :  null, "R2" : null, "R10" : null,"1R" :  null ,"2R" :  null }},
+        {"topic": "OUTPUT", "value": {"R0" : -1, "R00" : -1.0, "R1" :  -1.1, "R2" : -1.1,  "R10" :  -1.1, "1R" :  0.0 ,"2R" :  0.0 }},
+        {"topic": "OUTPUT", "value": {"R0" : -1, "R00" : -1.0, "R1" :  -1.5, "R2" : -1.5,  "R10" :  -1.5, "1R" :  0.0 ,"2R" :  0.0 }},
+        {"topic": "OUTPUT", "value": {"R0" : -2, "R00" : -2.0, "R1" :  -1.7, "R2" : -1.7,  "R10" :  -1.7, "1R" :  0.0 ,"2R" :  0.0 }},
+        {"topic": "OUTPUT", "value": {"R0" : 12346, "R00" : 12346.0, "R1" :  12345.7, "R2" : 12345.68,  "R10" :  12345.678, "1R" :  12350.0 ,"2R" :  12300.0 }},
+        {"topic": "OUTPUT", "value": {"R0" : -12346, "R00" : -12346.0, "R1" :  -12345.7, "R2" : -12345.68,  "R10" :  -12345.678, "1R" :  -12350.0 ,"2R" :  -12300.0 }}
+      ]
+    },
+    {
+      "name": "calculate ROUND with large DECIMAL values",
```

Comment:
Add post conditions to check the schema of `OUTPUT` has the scale and precision we're expecting, e.g 

```json
"post": {
        "sources": [
          {
            "name": "OUTPUT",
            "type": "stream",
            "valueSchema": "STRUCT<R0 DECIMAL(33, ?)...>"
          }
        ]
      }
```

Reply:
This is already tested in the expected output, e.g. a BigDecimal with scale 4 will match with expected output, e.g. '1234.0000', whereas same BigDecimal scale 0 will match with '1234'. I guess the test framework calls toString() on the BD.

Reply:
That's relying on the test framework correctly deserializing and comparing the outputs.  That might be working correctly at the moment, but could be broken in the future without us knowing.  A safer approach would be to assert that the _schema_ of the data is correct.

Reply:
done